inference-engine/thirdparty/ocv/opencv_hal_neon.hpp

```diff
@@ -2323,7 +2323,131 @@ inline v_uint16x8 v_mulhi(const v_uint16x8& a, uint16_t b) {
     return result;
 }
 
+CV_ALWAYS_INLINE v_int16x8 v_mulhrs(const v_int16x8& a, const v_int16x8& b) {
+    // Multiply
+    int32x4_t mul_lo = vmull_s16(vget_low_s16(a.val),
+                                 vget_low_s16(b.val));
+    int32x4_t mul_hi = vmull_s16(vget_high_s16(a.val),
+                                 vget_high_s16(b.val));
 
+    // Rounding narrowing shift right
+    // narrow = (int16_t)((mul + 16384) >> 15);
+    int16x4_t narrow_lo = vrshrn_n_s32(mul_lo, 15);
+    int16x4_t narrow_hi = vrshrn_n_s32(mul_hi, 15);
+
+    // Join together
+    return v_int16x8(vcombine_s16(narrow_lo, narrow_hi));
+}
+
+CV_ALWAYS_INLINE v_int16x8 v_mulhrs(const v_int16x8& a, short b) {
+    return v_mulhrs(a, v_setall_s16(b));
+}
+
+CV_ALWAYS_INLINE void v_gather_channel(v_uint8x16& vec, const uint8_t tmp[],
```

Comment:
why return value in this function is required as argument and in function below it is just a return value?

Reply:
Changed.